### PR TITLE
AA Signature

### PR DESCRIPTION
**Pull request by edjroz**
- [x]  Add Xion command to sign msgs using Abstract Accounts.
- [x] Add integration Test, demonstrating the signing process.

### CODE REVIEW

File: x/xion/client/cli/tx.go [hunk 1]
```diff
@@ -143,3 +157,144 @@ When using '--dry-run' a key name cannot be used, only a bech32 address.
...
+
+			if err := txBuilder.SetSignatures(sig); err != nil {
+				panic(err)
+			}
+
+			signBytes, err := clientCtx.TxConfig.SignModeHandler().GetSignBytes(signMode, signerData, txBuilder.GetTx())
+			if err != nil {
+				panic(err)
+			}
+			sigBytes, _, err := clientCtx.Keyring.Sign(clientCtx.GetFromName(), signBytes)
```
Comment (ash-burnt) [inline x/xion/client/cli/tx.go:227]: This doesn't include the authenticator index, how does this pass tests?

File: x/xion/client/cli/tx.go [hunk 2]
```diff
@@ -143,3 +157,144 @@ When using '--dry-run' a key name cannot be used, only a bech32 address.
...
+
+	if res.Account.TypeUrl != typeURL((*aatypes.AbstractAccount)(nil)) {
+		return nil, fmt.Errorf("signer %s is not an AbstractAccount", signerAddr.String())
+	}
+
+	var acc = &aatypes.AbstractAccount{}
+	if err = proto.Unmarshal(res.Account.Value, acc); err != nil {
+		return nil, err
+	}
+	//panic(fmt.Sprintf("Account: %s\nTypeURL: %s\nAcc: %+v", res.Account.TypeUrl, typeURL((*aatypes.AbstractAccount)(nil)), acc))
```
Comment (ash-burnt) [inline x/xion/client/cli/tx.go:293]: This can probably be removed

File: integration_tests/testdata/unsigned_msgs/bank_send_unsigned.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (ash-burnt) [inline integration_tests/testdata/unsigned_msgs/bank_send_unsigned.json:1]: this file seems unused. Is it referenced somewhere?

Comment (edjroz) [reply integration_tests/testdata/unsigned_msgs/bank_send_unsigned.json:1]: it's more for structure and doc reference as to where the Raw functions came from, we can remove the unsigned_msgs directories altogether

File: integration_tests/abstract_account_test.go [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+	require.NoError(t, err)
+
+	depositedFunds := fmt.Sprintf("%d%s", 100000, xion.Config().Denom)
+
+	// Register Abstract Account Using Public Key
+	registeredTxHash, err := ExecTx(t, ctx, xion.FullNodes[0],
+		xionUser.KeyName(),
+		"abstract-account", "register",
+		codeID,
+		fmt.Sprintf(`{"pubkey": "%s"}`, account["key"]),
```
Comment (ash-burnt) [inline integration_tests/abstract_account_test.go:91]: The definition for the init msg is 
```
#[cw_serde]
pub struct InstantiateMsg {
    pub id: u8,
    pub authenticator: Authenticator,
    pub signature: Binary,
}
```

not sure why this isnt panicking on deserialization

Comment (edjroz) [reply integration_tests/abstract_account_test.go:91]: Different contract. Will refactor to use this contract and msg.

**Review by ash-burnt — APPROVED** [review]

**Pull request merged**